WordPress/WordPressTest/MediaServiceRemoteRESTTests.swift

```diff
@@ -0,0 +1,144 @@
+import XCTest
+@testable import WordPress
+
+class MediaServiceRemoteRESTTests: XCTestCase {
+
+    let mockRemoteApi = MockWordPressComRestApi()
+    var mediaServiceRemote: MediaServiceRemoteREST!
+    let siteID = NSNumber(integer: 99999)
+
+    override func setUp() {
+        super.setUp()
+        mediaServiceRemote = MediaServiceRemoteREST(wordPressComRestApi: mockRemoteApi, siteID: siteID)
+    }
+
+    func mockRemoteMedia() -> RemoteMedia {
+
+        let remoteMedia = RemoteMedia()
+        remoteMedia.mediaID = 1
+        remoteMedia.postID = 2
+        remoteMedia.localURL = "http://www.wordpress.com"
+        remoteMedia.mimeType = "img/jpeg"
+        remoteMedia.file = "file_name"
+        return remoteMedia
+    }
+
+    func testGetMediaWithIDPath() {
+
+        let id = 1
+        let expectedPath = "v1.1/sites/\(siteID)/media/\(id)"
+        mediaServiceRemote.getMediaWithID(id, success: nil, failure: nil)
+        XCTAssertTrue(mockRemoteApi.getMethodCalled, "Wrong method")
+        XCTAssertEqual(mockRemoteApi.URLStringPassedIn, expectedPath, "Wrong path")
+    }
+
+    func testGetMediaWithID() {
+
+        let id = 1
+        let response = ["ID" : "2"]
+        var remoteMedia: RemoteMedia? = nil
+        mediaServiceRemote.getMediaWithID(id, success: {
+            remoteMedia = $0
```

Comment:
Ok, I've looked at this more. I'd love to see the test be stricter, so that it's more likely to fail if something is incorrect.

What do you think of adding a `XCTAssertEqual(remoteMedia.mediaID, id, …`?